common/expression/src/types/nullable.rs

```diff
@@ -188,10 +150,60 @@ impl<T: ArgType> ArgType for NullableType<T> {
     }
 
     fn create_builder(capacity: usize, generics: &GenericMap) -> Self::ColumnBuilder {
-        (
-            T::create_builder(capacity, generics),
-            MutableBitmap::with_capacity(capacity),
-        )
+        NullableColumnBuilder::with_capacity(capacity, generics)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct NullableColumn<T: ValueType> {
+    pub column: T::Column,
+    pub validity: Bitmap,
+}
+
+impl<T: ValueType> NullableColumn<T> {
+    pub fn len(&self) -> usize {
+        self.validity.len()
+    }
+
+    pub fn index(&self, index: usize) -> Option<Option<T::ScalarRef<'_>>> {
```

Comment:
Just one option is ok ?

Reply:
The first `Option` states whether the index is inbound and the second one states whether it's NULL.